src/gui/AutomationPatternView.cpp

```diff
@@ -474,18 +481,27 @@ void AutomationPatternView::scaleTimemapToFit( float oldMin, float oldMax )
 		return;
 	}
 
+	// TODO: Currently when rescaling the timeMap values to fit the new range of values (newMin and newMax)
+	// only the inValue is being considered and the outValue is being reset to the inValue (so discrete jumps
+	// are discarded). Possibly later we will want discrete jumps to be maintained so we will need to upgrade
+	// the logic to account for them.
```

Comment:
I can do that, but how exactly? Should I create a README file inside `src/gui` and write about this behavior there also keeping the TODO comment? Or you mean like a README section on the source file itself?

Reply:
If discrete jumps are discarded, and someone rescales, does it mess up their automation pattern? If yes, then this needs to be in release notes and have an issue open about it on merge. Otherwise, we should still have an issue open about it as a TODO item.

Reply:
I think we actually have to review the rescaling, just to make sure that's the behavior we want. Its current behavior is a little confusing, here's what I understood of it so far:

- `scaleTimemapToFit` is only called when we disconnect the last object from a pattern.
- It's called with `oldMin` and `oldMax` which correspond to the current min and max value of the pattern (which by itself correspond to the min and max of the first object connected to the pattern).
- Inside `scaleTimemapToFit`, `newMin` and `newMax` are set to the current min and max of the pattern, but since there's no pattern connected `firstObject` will actually return a dummy pattern with DEFAULT_MAX_VALUE and DEFAULT_MIN_VALUE (so after all it's kind of an overhead to call `getMin` and `getMax` since we will always get those values).
- The nodes are recalculated to fit in this new range.

I don't think getting rid of the discrete jumps will be too much of a hassle to the user for two reasons:
1) This is a very unique situation, where the user disconnects every connection from a pattern and then connects a new model to it, which doesn't happen very often.
2) The user will already have to deal with the nodes having the wrong value when he/she connects the new model again, because since it was rescaled to DEFAULT_MIN_VALUE and DEFAULT_MAX_VALUE the nodes will now have values between 0 and 1. So unless the new model's min and max values are 0 and 1 the user will already have "broke" the old pattern.

To be honest, the way it works right now it would probably be better to just remove all automation nodes when the last object is disconnected, not sure how useful the rescaling is at the moment. But to do that change first we will have to fix that issue with empty patterns not being valid. I agree with you this all have to be documented though, we can open an issue regarding this when the PR is merged!

Reply:
@IanCaio So what should we do with this for this PR?

Reply:
To be honest, I'd keep it as is for now. Compared to the current UX behavior, this doesn't make it worse: As mentioned, if the scaling method is called it's very likely the user will have a broken pattern once he connects a model again, so losing the discrete jumps is kind of meaningless compared to that.

I'd rather look at this issue separately, so I can come up with a proper solution and not reuse this PR for it. It's been a while I looked at this code, but from what I remember and without going much further on it maybe this would be a nice behavior:

1) Re-scaling every time you connect/disconnect a model, choosing as Min and Max the most restricted values amongst the models
2) As it does now, re-scaling to a default range when all models are disconnected
3) When connecting a model after disconnecting all of them, re-scale again to the range of that new model

Item 1 might sound like it would be weird, but truth is users rarely automate two parameters with different ranges in the same pattern. And if they do, it should be expected that the range will have to be enough to keep values valid for both models.